x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationSettings.java

```diff
@@ -32,274 +33,365 @@
-            "xpack.security.ssl.client_authentication", CLIENT_AUTH_SETTING_TEMPLATE);
-    public static final Function<String, Setting.AffixSetting<Optional<SSLClientAuth>>> CLIENT_AUTH_SETTING_REALM = realmType ->
-            Setting.affixKeySetting("xpack.security.authc.realms." + realmType + ".", "ssl.client_authentication",
-                    CLIENT_AUTH_SETTING_TEMPLATE);
-
-    private static final Function<String, Setting<Optional<VerificationMode>>> VERIFICATION_MODE_SETTING_TEMPLATE =
-            key -> new Setting<>(key, (String) null, s -> s == null ? Optional.empty() : Optional.of(VerificationMode.parse(s)),
-                    Property.NodeScope, Property.Filtered);
-    public static final Setting<Optional<VerificationMode>> VERIFICATION_MODE_SETTING_PROFILES = Setting.affixKeySetting(
-            "transport.profiles.", "xpack.security.ssl.verification_mode", VERIFICATION_MODE_SETTING_TEMPLATE);
-    public static final Function<String, Setting.AffixSetting<Optional<VerificationMode>>> VERIFICATION_MODE_SETTING_REALM = realmType ->
-            Setting.affixKeySetting("xpack.security.authc.realms." + realmType + ".", "ssl.verification_mode",
-                    VERIFICATION_MODE_SETTING_TEMPLATE);
+        key -> new Setting<>(
+            key,
+            (String) null,
+            s -> s == null ? Optional.empty() : Optional.of(SSLClientAuth.parse(s)),
+            Property.NodeScope,
+            Property.Filtered
+        );
+    private static final SslSetting<Optional<SSLClientAuth>> CLIENT_AUTH_SETTING = SslSetting.setting(
+        SslConfigurationKeys.CLIENT_AUTH,
+        CLIENT_AUTH_SETTING_TEMPLATE
+    );
+
+    private static final Function<String, Setting<Optional<VerificationMode>>> VERIFICATION_MODE_SETTING_TEMPLATE = key -> new Setting<>(
+        key,
+        (String) null,
+        s -> s == null ? Optional.empty() : Optional.of(VerificationMode.parse(s)),
+        Property.NodeScope,
+        Property.Filtered
+    );
+    private static final SslSetting<Optional<VerificationMode>> VERIFICATION_MODE = SslSetting.setting(
+        SslConfigurationKeys.VERIFICATION_MODE,
+        VERIFICATION_MODE_SETTING_TEMPLATE
+    );
+    public static final Function<String, Setting.AffixSetting<Optional<VerificationMode>>> VERIFICATION_MODE_SETTING_REALM =
+        VERIFICATION_MODE::realm;
 
     /**
      * @param prefix The prefix under which each setting should be defined. Must be either the empty string (<code>""</code>) or a string
      *               ending in <code>"."</code>
+     * @param acceptNonSecurePasswords Whether legacy (non-secure passwords should be accepted
      * @see #withoutPrefix
      * @see #withPrefix
      */
-    private SSLConfigurationSettings(String prefix) {
+    private SSLConfigurationSettings(String prefix, boolean acceptNonSecurePasswords) {
```

Comment:
Is acceptNonSecuredPassword introduced so that we can start ignoring the legacy settings without introducing any breaking change?

Reply:
Not quite.
It already existed as a concept in X509KeyPairSettings so that we could stop introducing "legacy" non-secure password settings for new stuff, like SAML signing keys.

As part of this work I had to pull it up to be a parameter in  `SSLConfigurationSettings`. I can't remember the exact reasons why it was necessary though.